### ISSUE TITLE
[feature request] mark own modules as not imported by default

### ISSUE THREAD

**Issue by pyrox0** [snowfallorg/lib#115]
I would like to be able to mark certain modules defined in my flake as not imported to all systems by default.

Currently, I have [a module](https://git.pyrox.dev/pyrox/nix/src/branch/main/modules/nixos/services/buildbot/default.nix) that configures a [buildbot-nix](https://github.com/nix-community/buildbot-nix) worker. However, this means that I also have to add `buildbot-nix.nixosModules.buildbot-worker` to `systems.modules.nixos` so that all systems pick it up. This increases evaluation time and is also a lot "dirtier" in terms of what I'm importing, since most machines will not have the buildbot worker on them. Therefore, if I could mark a module as non-importable somehow, I could make it so that only machines that will run a worker will import this, and by extension, also the buildbot-worker module from buildbot-nix. Is this possible at all with snowfall?

**Comment by pinkcreeper100** [comment]
iirc you can use
```nix
systems.hosts.my-host.modules = with inputs; [
  # my-input.nixosModules.my-module
];
```
to make specific systems import a module

**Comment by pinkcreeper100** [comment]
wait you're already using that 🥴

**Comment by pyrox0** [comment]
Yes, what I want to do is to make it so that `self.nixosModules."services/buildbot"`(which is defined in my flake) is not imported to all systems by default(which happens with all systems by default, per the first line of the [Modules doc page](https://snowfall.org/guides/lib/modules/)), as it then requires me to import `buildbot-nix.nixosModules.buildbot-worker` to all systems.

**Comment by pinkcreeper100** [comment]
you could try - although it would be a bit awkward - to move the module to a separate repo and import from there
